lib/aurora_bootstrapper/table.rb

```diff
@@ -1,11 +1,16 @@
+require 'aws-sdk-s3'
+
 module AuroraBootstrapper
   class Table
-    def initialize( database_name:, table_name:, client:, blacklisted_fields: [] )
+    def initialize( database_name:, table_name:, client:, blacklisted_fields: [], s3_client: nil )
       @blacklisted_fields = blacklisted_fields
       @database_name      = database_name
       @table_name         = table_name
       @client             = client
-      @export_date        = ENV.fetch( 'EXPORT_DATE', nil )
+      d                   = DateTime.now
+      d_str               = d.strftime("%Y-%m-%d")
+      @export_date        = ENV.fetch( 'EXPORT_DATE', d_str )
+      @notifier           = Notifier.new s3_client: s3_client
```

Comment:
does this have to come from the outside or can this be defined inside the class?
```ruby
class Table
  def notifier
    @notifier||= Notifier.new
  end
end

class Notifier
  def initialize
    region  = ENV.fetch( 'REGION', 'us-west-2')
    @client = Aws::S3::Client.new(region: region)
  end
end
```

Reply:
Only create s3 client inside Notifier file; move the initialization of notifier to exporter file and use notifier instance inside exporter.rb outside of databases.name loop